File: src/charting/data/RadarData.java

package com.github.mikephil.charting.data;

import com.github.mikephil.charting.highlight.Highlight;
import com.github.mikephil.charting.interfaces.datasets.IRadarDataSet;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * Data container for the RadarChart.
 *
 * @author Philipp Jahoda
 */
public class RadarData extends ChartData<IRadarDataSet> {

    private List<String> this.mLabels;

    public RadarData() {
        super();
    }

    public RadarData(List<IRadarDataSet> dataSets) {
        super(dataSets);
    }

    public RadarData(IRadarDataSet... dataSets) {
        super(dataSets);
    }

    /**
     * Sets the labels that should be drawn around the RadarChart at the end of each web line.
     *
     * @param labels
     */
    public setLabels(List<String> labels) {
        this.mLabels = labels;
    }

    /**
     * Sets the labels that should be drawn around the RadarChart at the end of each web line.
     *
     * @param labels
     */
    public setLabels(String... labels) {
        this.mLabels = Arrays.asList(labels);
    }

    public List<String> getLabels() {
        return this.mLabels;
    }

    
    public Entry getEntryForHighlight(Highlight highlight) {
        return getDataSetByIndex(highlight.getDataSetIndex()).getEntryForIndex( highlight.getX());
    }
}

File: src/charting/data/RadarData.ts
import { Highlight } from '../highlight/Highlight';
import { ChartData } from './ChartData';
import { RadarDataSet } from './RadarDataSet';
import { RadarEntry } from './RadarEntry';

/**
 * Data container for the RadarChart.
 *

 */
export class RadarData extends ChartData<RadarEntry, RadarDataSet> {
    /**
     * Sets the labels that should be drawn around the RadarChart at the end of each web line.
     */
    labels: string[];

    constructor(dataSets: RadarDataSet[]) {
        super(dataSets);
    }

    public getEntryForHighlight(highlight: Highlight) {
        return this.getDataSetByIndex(highlight.dataSetIndex).getEntryForIndex(highlight.x);
    }
}
